### ISSUE TITLE
Support Sub Module release strategy

### ISSUE THREAD

**Issue by salameer** [Azure/azure-sdk-for-go#831]

**Comment by marstr** [comment]
Adding context:

Some folks from the Azure Storage team have taken an interest in Go. They've done a tremendous amount of work in partnership with our team to improve the [Autorest Go generator](https://github.com/Azure/autorest.go) to accommodate their needs. As part of that effort, they will also be making hand-written convenience layers over the top of what is generated. To best facilitate their efforts, they've decided to work and release at their own cadence in a repository outside of the Azure-SDK-for-Go.

However, we'd still like this repository to be a one stop shop for folks looking to point their Go program at Microsoft's Cloud. To achieve this, we want to vendor stable bits from their repositories here. There is some prior-art here, in terms of doing this using `git submodule`. That prior art however, failed due to unforeseen complications with package-management tools (#581). Those reasons are unresolved, and sort of an intrinsic part of using git for distributing source. We are not stumped though, as we can use a `git subtree` instead. That still needs to be tested, but I believe it should work better because there is less machinery that Git needs to manipulate.

**Comment by tombuildsstuff** [comment]
@marstr is the intention is to deprecate the current Storage SDK in favour of the one in the separate repository?

**Comment by jhendrixMSFT** [comment]
@tombuildsstuff Short answer is yes, however it will come in phases.  The current storage SDK contains code for files, blobs, queues and tables.  We have a new blob storage SDK (azblob) so the existing blob storage code will be deprecated, and as files, queues and tables are completed the existing ones will be deprecated as well.

https://github.com/Azure/azure-storage-blob-go
https://godoc.org/github.com/Azure/azure-storage-blob-go/2016-05-31/azblob

**Comment by marstr** [comment]
After meeting with @salameer, @jhendrixMSFT, and @JeffreyRichter this morning we've decided to go a slightly different direction.

Now instead of using `git` to solve these troubles, instead we'll publish packages that were developed in an external repository in the form of aliases, the same way we do the contents of the `profiles` directory. Doing this will get us out of some dependency management frustrations, as well as allow more flexibility in terms of how folks consume packages. I've closed #855, and will submit another PR with the aliased storage packages.
